public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx

```diff
@@ -128,10 +131,12 @@ export class DetectionVisualEditor extends React.Component<
       this.props.onChange(this.createDetectionYml());
     }
 
-    if (Object.keys(this.state.errors.fields).length || !this.validateValuesExist()) {
-      this.props.setIsDetectionInvalid(true);
-    } else {
-      this.props.setIsDetectionInvalid(false);
+    const isValid = !!Object.keys(this.state.errors.fields).length || !this.validateValuesExist();
```

Comment:
Any reason to make this check inside componentDidUpdate? Can we do this where the state is actually updated?
We will update the prop in parent here and then trigger this again?

Reply:
Yeah, the reason here is that some fields are not appropriately validated and are not marked as invalid by formik, so if the form detection is validated I would trigger the fields validation in the detection editor so that I am sure the fields are marked properly.